public/app/features/explore/TraceView/components/TraceTimelineViewer/SpanLinks.tsx

```diff
@@ -2,106 +2,44 @@ import { css } from '@emotion/css';
-              onClick={
-                link.onClick
-                  ? (event) => {
-                      reportInteraction('grafana_traces_trace_view_span_link_clicked', {
-                        datasourceType: datasourceType,
-                        grafana_version: config.buildInfo.version,
-                        type: 'log',
-                        location: 'menu',
-                      });
-                      event?.preventDefault();
-                      link.onClick!(event);
-                      closeMenu();
-                    }
-                  : undefined
-              }
-              url={link.href}
-              className={styles.menuItem}
-            />
-          ))}
-        </MenuGroup>
-      ) : null}
-      {!!links.metricLinks?.length ? (
-        <MenuGroup label="Metrics">
-          {links.metricLinks.map((link, i) => (
-            <MenuItem
-              key={i}
-              label={link.title ?? 'Metrics for this span'}
-              onClick={
-                link.onClick
-                  ? (event) => {
-                      reportInteraction('grafana_traces_trace_view_span_link_clicked', {
-                        datasourceType: datasourceType,
-                        grafana_version: config.buildInfo.version,
-                        type: 'metric',
-                        location: 'menu',
-                      });
-                      event?.preventDefault();
-                      link.onClick!(event);
-                      closeMenu();
-                    }
-                  : undefined
-              }
-              url={link.href}
-              className={styles.menuItem}
-            />
-          ))}
-        </MenuGroup>
-      ) : null}
-      {!!links.traceLinks?.length ? (
-        <MenuGroup label="Traces">
-          {links.traceLinks.map((link, i) => (
-            <MenuItem
-              key={i}
-              label={link.title ?? 'View linked span'}
-              onClick={
-                link.onClick
-                  ? (event) => {
-                      reportInteraction('grafana_traces_trace_view_span_link_clicked', {
-                        datasourceType: datasourceType,
-                        grafana_version: config.buildInfo.version,
-                        type: 'trace',
-                        location: 'menu',
-                      });
-                      event?.preventDefault();
-                      link.onClick!(event);
-                      closeMenu();
-                    }
-                  : undefined
-              }
-              url={link.href}
-              className={styles.menuItem}
-            />
-          ))}
-        </MenuGroup>
-      ) : null}
-    </>
-  );
+  return links.map((link, i) => (
+    <MenuItem
+      key={i}
+      label={link.title || 'Link'}
+      onClick={
+        link.onClick
+          ? (event) => {
+              reportInteraction(`grafana_traces_${link.type}_view_span_link_clicked`, {
```

Comment:
We also use it here for tracking

Reply:
Love this, so much tidier :) 

Could you please update this to [something like](https://github.com/grafana/grafana/blob/main/public/app/features/explore/TraceView/components/TraceTimelineViewer/SpanLinks.tsx#L31)

`reportInteraction('grafana_traces_trace_view_span_link_clicked', {
    datasourceType: datasourceType,
    grafana_version: config.buildInfo.version,
    type: link.type,
    location: 'menu'
  });`
                      
All span link clicks are tracking to `grafana_traces_trace_view_span_link_clicked` and we specify the type in the object we send with that event. This is so it fits in nicely with existing tracking and so we don't have separate events  for each type of click.

Reply:
Fixed!